uc-crux-llvm/src/UCCrux/LLVM/FullType/Type.hs

```diff
@@ -0,0 +1,279 @@
+
+data IsPtrRepr m ft = forall ft'. IsPtrRepr (ft :~: 'FTPtr ft')
+
+isPtrRepr :: forall m ft. FullTypeRepr m ft -> Maybe (IsPtrRepr m ft)
+isPtrRepr =
+  \case
+    FTPtrRepr _ -> Just (IsPtrRepr Refl)
+    _ -> Nothing
+
+$(return [])
+
+-- | We assume (via unsafeCoerce) that types with the same L.Ident are the same.
+-- This is validated by the existential used in @makeModuleTypes@.
+instance TestEquality (PartTypeRepr m) where
+  testEquality =
+    $( U.structuralTypeEquality
+         [t|PartTypeRepr|]
+         ( let appAny con = U.TypeApp con U.AnyType
+            in [ ( appAny (appAny (U.ConType [t|FullTypeRepr|])),
+                   [|testEquality|]
+                 ),
+                 ( appAny (appAny (U.ConType [t|Ctx.Assignment|])),
+                   [|testEquality|]
+                 ),
+                 ( appAny (U.TypeApp (U.ConType [t|Const|]) (U.ConType [t|L.Ident|])),
+                   [|
+                     \(Const ident1 :: Const L.Ident ft1) (Const ident2 :: Const L.Ident ft2) ->
+                       if ident1 == ident2 then Just (unsafeCoerce Refl :: ft1 :~: ft2) else Nothing
+                     |]
+                 )
+               ]
+         )
+     )
+
+instance TestEquality (FullTypeRepr m) where
+  testEquality =
+    $( U.structuralTypeEquality
+         [t|FullTypeRepr|]
+         ( let appAny con = U.TypeApp con U.AnyType
+            in [ ( appAny (U.ConType [t|NatRepr|]),
+                   [|testEquality|]
+                 ),
+                 ( appAny (U.ConType [t|CrucibleTypes.FloatInfoRepr|]),
+                   [|testEquality|]
+                 ),
+                 ( appAny (appAny (U.ConType [t|FullTypeRepr|])),
+                   [|testEquality|]
+                 ),
+                 ( appAny (appAny (U.ConType [t|PartTypeRepr|])),
+                   [|testEquality|]
+                 ),
+                 ( appAny (appAny (U.ConType [t|Ctx.Assignment|])),
+                   [|testEquality|]
+                 )
+               ]
+         )
+     )
+
+-- | See note on 'TestEquality' instance.
+instance OrdF (PartTypeRepr m) where
+  compareF =
+    $( U.structuralTypeOrd
+         [t|PartTypeRepr|]
+         ( let appAny con = U.TypeApp con U.AnyType
+            in [ ( appAny (appAny (U.ConType [t|FullTypeRepr|])),
+                   [|compareF|]
+                 ),
+                 ( appAny (appAny (U.ConType [t|Ctx.Assignment|])),
+                   [|compareF|]
+                 ),
+                 ( appAny (U.TypeApp (U.ConType [t|Const|]) (U.ConType [t|L.Ident|])),
+                   [|
+                     \(Const ident1 :: Const L.Ident ft1) (Const ident2 :: Const L.Ident ft2) ->
+                       case compare ident1 ident2 of
+                         LT -> unsafeCoerce LTF :: OrderingF ft1 ft2
+                         GT -> unsafeCoerce GTF :: OrderingF ft1 ft2
+                         EQ -> unsafeCoerce EQF :: OrderingF ft1 ft2
```

Comment:
Similar issue here, I don't trust this unsafe coerce.

Reply:
The relevant constructors are no longer exported!